### PR TITLE
Add java.time.Duration support in the JavaDSL in akka-actor module #24646

### PR DESCRIPTION
**Pull request by patriknw**
Continued https://github.com/akka/akka/pull/24735

* Added deprecations where it was only in Java API.
* Changed signature of `retry` since that hasn't been released yet

Refs #24646

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/AbstractFSM.scala [hunk 1]
```diff
@@ -430,7 +430,7 @@ abstract class AbstractFSM[S, D] extends FSM[S, D] {
    */
   final def setTimer(name: String, msg: Any, timeout: java.time.Duration): Unit = {
     import JavaDurationConverters._
-    setTimer(name, msg, timeout.asScala)
+    setTimer(name, msg, timeout.asScala, false)
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/actor/AbstractFSM.scala]: Any reason for this?

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java [hunk 2]
```diff
@@ -412,7 +412,7 @@ public void usePatternsGracefulStop() throws Exception {
     //#gracefulStop
     try {
       CompletionStage<Boolean> stopped =
-        gracefulStop(actorRef, Duration.create(5, TimeUnit.SECONDS), Manager.SHUTDOWN);
+        gracefulStop(actorRef, java.time.Duration.ofSeconds(5), Manager.SHUTDOWN);
```
Comment (chbatey) [inline akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: There are still a lot of things in this test (so in the docs) that use scala's Duration e.g. setting receiveTimeout.

We ofc don't need to get to them all in this PR

**Review by chbatey — APPROVED** [review]
All the changes look good.

File: akka-docs/src/test/java/jdocs/actor/ActorDocTest.java (hunk 2 above)
Comment (patriknw) [reply akka-docs/src/test/java/jdocs/actor/ActorDocTest.java]: Yes, we will not close the ticket #24646, please add notes there. I think receiveTimeout was already mentioned.

**Comment by patriknw** [comment]
I reviewed the other PR thoroughly, so merging

**Pull request merged**